pkg/resource/table/hooks.go

```diff
@@ -140,139 +140,164 @@ func (rm *resourceManager) customUpdateTable(
+		if err := rm.syncTable(ctx, desired, delta); err != nil {
+			return nil, err
+		}
+	case delta.DifferentAt("Spec.ProvisionedThroughput"):
+		if err := rm.syncTableProvisionedThroughput(ctx, desired); err != nil {
+			return nil, err
+		}
+
+	case delta.DifferentAt("Spec.GlobalSecondaryIndexes"):
+		if err := rm.syncTableGlobalSecondaryIndexes(ctx, latest, desired); err != nil {
+			return nil, err
+		}
+	}
+
 	return &resource{ko}, nil
 }
 
-// syncTableTags updates a dynamodb table tags.
-//
-// TODO(hilalymh): move this function to a common utility file. This function can be reused
-// to tag GlobalTable resources.
-func (rm *resourceManager) syncTableTags(
+// syncTableProvisionedThroughput updates a given table provisioned throughputs
+func (rm *resourceManager) syncTableProvisionedThroughput(
 	ctx context.Context,
-	latest *resource,
-	desired *resource,
+	r *resource,
 ) (err error) {
 	rlog := ackrtlog.FromContext(ctx)
-	exit := rlog.Trace("rm.syncTableTags")
+	exit := rlog.Trace("rm.syncTableProvisionedThroughput")
 	defer exit(err)
 
-	added, updated, removed := computeTagsDelta(latest.ko.Spec.Tags, desired.ko.Spec.Tags)
-
-	// There are no API calls to update an existing tag. To update a tag we will have to first
-	// delete it and then recreate it with the new value.
-
-	// Tags to remove
-	for _, updatedTag := range updated {
-		removed = append(removed, updatedTag.Key)
+	input := &svcsdk.UpdateTableInput{
+		TableName:             aws.String(*r.ko.Spec.TableName),
+		ProvisionedThroughput: &svcsdk.ProvisionedThroughput{},
 	}
-	// Tags to create
-	added = append(added, updated...)
-
-	if len(removed) > 0 {
-		_, err = rm.sdkapi.UntagResourceWithContext(
-			ctx,
-			&svcsdk.UntagResourceInput{
-				ResourceArn: (*string)(latest.ko.Status.ACKResourceMetadata.ARN),
-				TagKeys:     removed,
-			},
-		)
-		rm.metrics.RecordAPICall("GET", "UntagResource", err)
-		if err != nil {
-			return err
+	if r.ko.Spec.ProvisionedThroughput != nil {
+		if r.ko.Spec.ProvisionedThroughput.ReadCapacityUnits != nil {
+			input.ProvisionedThroughput.ReadCapacityUnits = aws.Int64(*r.ko.Spec.ProvisionedThroughput.ReadCapacityUnits)
+		} else {
+			input.ProvisionedThroughput.ReadCapacityUnits = aws.Int64(0)
 		}
-	}
 
-	if len(added) > 0 {
-		_, err = rm.sdkapi.TagResourceWithContext(
-			ctx,
-			&svcsdk.TagResourceInput{
-				ResourceArn: (*string)(latest.ko.Status.ACKResourceMetadata.ARN),
-				Tags:        sdkTagsFromResourceTags(added),
-			},
-		)
-		rm.metrics.RecordAPICall("GET", "UntagResource", err)
-		if err != nil {
-			return err
+		if r.ko.Spec.ProvisionedThroughput.WriteCapacityUnits != nil {
+			input.ProvisionedThroughput.WriteCapacityUnits = aws.Int64(*r.ko.Spec.ProvisionedThroughput.WriteCapacityUnits)
+		} else {
+			input.ProvisionedThroughput.WriteCapacityUnits = aws.Int64(0)
 		}
+	} else {
+		input.ProvisionedThroughput.ReadCapacityUnits = aws.Int64(0)
+		input.ProvisionedThroughput.WriteCapacityUnits = aws.Int64(0)
```

Comment:
Oh jeez please tell me this isn't possible? lol

Reply:
It is not lol, but i think if a user deletes these fields that what the controller would be expected to do?